python/ray/tests/test_generators.py

```diff
@@ -336,6 +336,32 @@ def static(num_returns):
                 ray.get(ref)
 
 
+def test_dynamic_generator_gc_each_yield(ray_start_cluster):
+    # Need to shutdown when going from ray_start_regular_shared to ray_start_cluster
```

Comment:
this is ok here, but in general I'm thinking we should isolate all tests that can use a shared fixture in a single file and don't mix them, else things like this will get confusing/error prone